src/commands/producers.ts

```diff
@@ -67,35 +68,69 @@ export class ProduceRecordCommandHandler {
             };
         });
 
-        const producer = await client.producer();
-        await producer.connect();
-
-        channel.appendLine(`Producing record(s)`);
-        const startOperation = performance.now();
-
-        try {
-            await producer.send({
-                topic: topicId,
-                messages: messages,
-            });
-
-
-            const finishedOperation = performance.now();
-            const elapsed = (finishedOperation - startOperation).toFixed(2);
-
-            channel.appendLine(`Produced ${times} record(s) (${elapsed}ms)`);
-
-            this.explorer.refresh();
-        } catch (error) {
-            const finishedOperation = performance.now();
-            const elapsed = (finishedOperation - startOperation).toFixed(2);
-            channel.appendLine(`Failed to produce record(s) (${elapsed}ms)`);
+        command.clusterId = client.cluster.id;
+        const producerUri = createProducerUri(command);
+        const record = {
+            topic: topicId,
+            messages: messages,
+        };
+        // Start the producer
+        await startProducerWithProgress(producerUri, record, this.producerCollection, channel, times, this.explorer);
+    }
+}
 
-            if (error.message) {
-                channel.appendLine(`Error: ${error.message}`);
-            } else {
-                channel.appendLine(`Error: ${error}`);
-            }
-        }
+async function startProducerWithProgress(producerUri: vscode.Uri, record: ProducerRecord, producerCollection: ProducerCollection, channel: vscode.OutputChannel, times: number, explorer?: KafkaExplorer) {
+    const producer = producerCollection.get(producerUri);
+    if (producer && producer.state === ProducerLaunchState.connecting) {
```

Comment:
isBusy(producer.state)

need to check all non-idle states

Reply:
fixed